### PR TITLE
fix: Deprecations and Errors

### PR DESCRIPTION
**Pull request by maharshi-cd**
## what
* Removed deprecations and errors 
* Added  azuread_administrator block as its resource is not supported 
* Added user assigned identity
* Replaced deprecated arguments with new one

## why
* Schedule maintenance

### CODE REVIEW

File: .github/workflows/auto_assignee.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Auto Assign PRs
+
+on:
+  pull_request:
+    types: [opened, reopened]
+
+  workflow_dispatch:
+jobs:
+  assignee:
+    uses: clouddrove/github-shared-workflows/.github/workflows/auto_assignee.yml@1.2.8
```
Comment (archit-cd) [inline .github/workflows/auto_assignee.yml]: we have to refer to master, not tag.

File: .github/workflows/automerge.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+name: Auto merge
+on:
+  pull_request:
+jobs:
+  auto-merge:
+    uses: clouddrove/github-shared-workflows/.github/workflows/auto_merge.yml@1.2.8
```
Comment (archit-cd) [inline .github/workflows/automerge.yml]: we have to refer to master, not tag.

File: .github/workflows/changelog.yml [hunk 3]
```diff
@@ -7,7 +7,7 @@ on:
   workflow_dispatch:
 jobs:
   changelog:
-    uses: clouddrove/github-shared-workflows/.github/workflows/changelog.yml@master
+    uses: clouddrove/github-shared-workflows/.github/workflows/changelog.yml@1.2.8
```
Comment (archit-cd) [inline .github/workflows/changelog.yml]: we have to refer to master, not tag.

File: .github/workflows/tf-checks.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+name: tf-checks
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+  workflow_dispatch:
+jobs:
+  complete-example:
+    uses: clouddrove/github-shared-workflows/.github/workflows/tf-checks.yml@1.2.8
```
Comment (archit-cd) [inline .github/workflows/tf-checks.yml]: we have to refer to master, not tag.

File: .github/workflows/tflint.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+name: tf-lint
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+  workflow_dispatch:
+jobs:
+  tf-lint:
+    uses: clouddrove/github-shared-workflows/.github/workflows/tf-lint.yml@1.2.8
```
Comment (archit-cd) [inline .github/workflows/tflint.yml]: we have to refer to master, not tag.

File: .github/workflows/tfsec.yml [hunk 6]
```diff
@@ -5,7 +5,7 @@ on:
   workflow_dispatch:
 jobs:
   tfsec:
-    uses: clouddrove/github-shared-workflows/.github/workflows/tfsec.yml@master
+    uses: clouddrove/github-shared-workflows/.github/workflows/tfsec.yml@1.2.8
```
Comment (archit-cd) [inline .github/workflows/tfsec.yml]: we have to refer to master, not tag.

File: examples/Simple_SQL_Single_Database_with_Private_Endpoint/main.tf [hunk 7]
```diff
@@ -1,7 +1,10 @@
 provider "azurerm" {
   features {}
+  subscription_id = "01111111111110-11-11-11-11"
```
Comment (archit-cd) [inline examples/Simple_SQL_Single_Database_with_Private_Endpoint/main.tf]: This is not the standard that we decided, please update this.

File: main.tf [hunk 8]
```diff
@@ -244,83 +265,56 @@ resource "azurerm_mssql_server_vulnerability_assessment" "va_secondary" {
...
+  partner_server {
+    id = azurerm_mssql_server.secondary[0].id
   }
 
   read_write_endpoint_failover_policy {
     mode          = "Automatic"
     grace_minutes = 60
   }
-
-  readonly_endpoint_failover_policy {
```
Comment (archit-cd) [inline main.tf:316]: Is there no alternative for this?

File: versions.tf [hunk 9]
```diff
@@ -1,13 +1,21 @@
...
-  required_version = ">= 1.0.0"
+  required_version = ">=1.6.6"
 }
 
 terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = ">=2.90.0"
+      version = ">=3.112.0"
```
Comment (archit-cd) [inline versions.tf:10]: Are you sure that it requires `>=112`?

File: versions.tf [hunk 10]
```diff
@@ -1,13 +1,21 @@
...
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "~> 2.0"
+    }
+    null = {
+      source  = "hashicorp/null"
+      version = "~> 2.0"
```
Comment (archit-cd) [inline versions.tf]: Can we use `>=` in both random and null provider?

**Review by archit-cd — CHANGES_REQUESTED** [review]
Please resolve all comments.

File: .github/workflows/auto_assignee.yml (hunk 1 above)
Comment (maharshi-cd) [reply .github/workflows/auto_assignee.yml]: done

File: .github/workflows/automerge.yml (hunk 2 above)
Comment (maharshi-cd) [reply .github/workflows/automerge.yml]: done

File: .github/workflows/changelog.yml (hunk 3 above)
Comment (maharshi-cd) [reply .github/workflows/changelog.yml]: done

File: examples/Simple_SQL_Single_Database_with_Private_Endpoint/main.tf (hunk 7 above)
Comment (maharshi-cd) [reply examples/Simple_SQL_Single_Database_with_Private_Endpoint/main.tf]: updated with new one

File: main.tf (hunk 8 above)
Comment (maharshi-cd) [reply main.tf:316]: this is deprecated
used this readonly_endpoint_failover_policy_enabled

File: versions.tf (hunk 9 above)
Comment (maharshi-cd) [reply versions.tf:10]: yess 100 percent sure

File: versions.tf (hunk 10 above)
Comment (maharshi-cd) [reply versions.tf]: yess have done that

File: .github/workflows/tfsec.yml (hunk 6 above)
Comment (maharshi-cd) [reply .github/workflows/tfsec.yml]: done

**Pull request merged**